src/System.Private.CoreLib/shared/System/Number.BigInteger.cs

```diff
@@ -849,6 +902,12 @@ public static void Multiply(ref BigInteger lhs, ref BigInteger rhs, ref BigInteg
                 }
             }
 
+            public static void Pow2(uint exponent, out BigInteger result)
```

Comment:
I see this is following the same pattern as `Pow10` below, but it seems like an odd method signature - `void` return and a single `out` parameter. Is there any advantages to this vs. just returning the `BigInteger`?

Reply:
I was mainly following the existing code-pattern. The primary benefit, however, would be that you can reuse an existing `BigInteger` local rather than having to allocate a new one.

This can help keep the stack-space smaller given how large each `BigInteger` local is.